graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/SetPrototypeBuiltins.java

```diff
@@ -87,7 +101,14 @@ protected SetPrototypeBuiltins() {
         has(1),
         forEach(1),
         values(0),
-        entries(0);
+        entries(0),
+        union(1),
+        intersection(1),
+        difference(1),
+        symmetricDifference(1),
+        isSubsetOf(1),
+        isSupersetOf(1),
+        isDisjointedFrom(1);
```

Comment:
You need to make sure those entries are only added in ECMAScript-2022 mode. For that, please add the following function after the `getLength` function, around Line 125:

```
@Override
        public int getECMAScriptVersion() {
            if (EnumSet.of(union, intersection, difference, symmetricDifference, isSubsetOf, isSupersetOf, isDisjointedFrom).contains(this)) {
                return JSConfig.ECMAScript2022;
            }
            return BuiltinEnum.super.getECMAScriptVersion();
        }
```

After that, those methods are only available in ECMAScript 2022 or later. This means, you have to start with the additional flag of `--js.ecmascript-version=2022`. This also applies to your tests, you will have to modify them to use:

```
JSTest.newContextBuilder().option(JSContextOptions.ECMASCRIPT_VERSION_NAME, "2022").build()
```